src/main/java/net/dv8tion/jda/core/requests/restaction/order/CategoryOrderAction.java

```diff
@@ -0,0 +1,84 @@
+
+package net.dv8tion.jda.core.requests.restaction.order;
+
+import net.dv8tion.jda.core.entities.*;
+import net.dv8tion.jda.core.utils.Checks;
+
+import java.util.List;
+
+/**
+ * An extension of {@link net.dv8tion.jda.core.requests.restaction.order.ChannelOrderAction ChannelOrderAction} with
+ * similar functionality, but constrained to the bounds of a single {@link net.dv8tion.jda.core.entities.Category Category}.
+ * <br>To apply the changes you must finish the {@link net.dv8tion.jda.core.requests.RestAction RestAction}.
+ *
+ * <p>Before you can use any of the {@code move} methods
+ * you must use either {@link #selectPosition(Object) selectPosition(Channel)} or {@link #selectPosition(int)}!
+ *
+ * @param  <T>
+ *         The type of {@link net.dv8tion.jda.core.entities.Channel Channel} to move
+ *         using this CategoryOrderAction, either {@link net.dv8tion.jda.core.entities.TextChannel TextChannel},
+ *         or {@link net.dv8tion.jda.core.entities.VoiceChannel VoiceChannel}.
+ *
+ * @author Kaidan Gustave
+ */
+public class CategoryOrderAction<T extends Channel> extends ChannelOrderAction<T>
+{
+    protected final Category category;
+
+    /**
+     * Creates a new CategoryOrderAction for the specified {@link net.dv8tion.jda.core.entities.Category Category}
+     *
+     * @param  category
+     *         The target {@link net.dv8tion.jda.core.entities.Category Category}
+     *         which the new CategoryOrderAction will order channels from.
+     * @param  type
+     *         The {@link net.dv8tion.jda.core.entities.ChannelType ChannelType} that
+     *         matches the returning value of {@link net.dv8tion.jda.core.entities.Channel#getType() Channel#getType()}
+     *         for the generic {@link net.dv8tion.jda.core.entities.Channel Channel} type {@code T}.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the {@code ChannelType} is not one that can be retrieved from a {@code Category}.
+     *         Currently the only two allowed are {@link ChannelType#TEXT} and {@link ChannelType#VOICE}.
+     */
+    public CategoryOrderAction(Category category, ChannelType type)
+    {
+        super(category.getGuild(), type,
+            (List<T>)(type == ChannelType.TEXT ? category.getTextChannels() : category.getVoiceChannels()));
+        this.category = category;
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.entities.Category Category}
+     * controlled by this CategoryOrderAction.
+     *
+     * @return The {@link net.dv8tion.jda.core.entities.Category Category}
+     *         of this CategoryOrderAction.
+     */
+    public Category getCategory()
+    {
+        return category;
+    }
+
+    @Override
+    protected void validateInput(T entity)
+    {
+        Checks.check(entity.getParent() != null, "Provided channel must be nested in a Category!");
+        Checks.check(entity.getParent().equals(getCategory()), "Provided channel's Category is not this Category!");
```

Comment:
This can be simplified to `Checks.check(category.equals(entity.getParent());`
Instead of null-checking the parent. However a null-check for the entity itself is required.